buffer/src/test/java/io/netty/buffer/CompositeByteBufTest.java

```diff
@@ -0,0 +1,24 @@
+package io.netty.buffer;
+
+import io.netty.util.ResourceLeakDetector;
+import org.junit.Test;
+
+import static io.netty.buffer.Unpooled.*;
+import static org.junit.Assert.*;
+
+public class CompositeByteBufTest {
+
+    @Test
+    public void testAddComponentWithLeakAwareByteBuf() {
+        ResourceLeakDetector.setLevel(ResourceLeakDetector.Level.PARANOID);
```

Comment:
@jingene we must reset this to the previous level once this test completes.

Reply:
I see. fixed(https://github.com/netty/netty/pull/9398/commits/8da1276cfd719d61b717983384dd1139b1e6cf9a). Thanks.

Reply:
Didn't you also adjust the sampling interval before ?

Reply:
Also I wonder if we couldn't construct the buffer directly and so not need all of the above. Please check our `*LeakAwareByteBufTest` classes.

Reply:
@normanmaurer Thank you for your kind guide. ```AdvancedLeakAwareByteBufTest#wrap``` guarantees return ```AdvancedLeakAwareByteBuf``` instance so I removed ```ResourceLeakDetector``` manipulation codes (https://github.com/netty/netty/pull/9398/commits/c479b1064934be4d3023657b64d2a7cb5cb3aadb)